spirv_cross.cpp

```diff
@@ -417,7 +417,7 @@ bool Compiler::is_immutable(uint32_t id) const
 		auto &var = get<SPIRVariable>(id);
 
 		// Anything we load from the UniformConstant address space is guaranteed to be immutable.
-		bool pointer_to_const = var.storage == StorageClassUniformConstant;
+		bool pointer_to_const = var.storage == StorageClassUniformConstant || var.storage == StorageClassUniform;
```

Comment:
This doesn't seem right. Why is StorageClassUniform added here?

Reply:
I think it missed nested struct unpacking but I didn't try running without that code after merging latest head. will double check